common/models/user.js

```diff
@@ -558,6 +573,13 @@ module.exports = function(User) {
       return cb.promise;
     }
 
+    if (options.password && options.password.length > MAX_PASSWORD_LENGTH) {
```

Comment:
I think we should call `User.validatePassword`, so that we don't duplicate the validation logic.